haystack/nodes/prompt/prompt_node.py

```diff
@@ -840,19 +849,48 @@ def is_supported_template(self, prompt_template: Union[str, PromptTemplate]) ->
         template_name = prompt_template if isinstance(prompt_template, str) else prompt_template.name
         return template_name in self.prompt_templates
 
-    def get_prompt_template(self, prompt_template: Union[str, PromptTemplate, None]) -> PromptTemplate:
+    def get_prompt_template(self, prompt_template: Union[str, PromptTemplate, None] = None) -> Optional[PromptTemplate]:
         """
-        Returns a prompt template by name.
-        :param prompt_template_name: The name of the prompt template to be returned.
-        :return: The prompt template object.
+        Resolves a prompt template.
+
+        :param prompt_template: The prompt template to be resolved. You can choose between the following types:
+            - None: returns the default prompt template
+            - PromptTemplate: returns the given prompt template object
+            - str: parses the string depending on its content:
+                - prompt template name: returns the prompt template registered with the given name
+                - prompt template yaml: returns a prompt template specified by the given yaml
+                - prompt text: returns a copy of the default prompt template with the given prompt text
+
+            :return: The prompt template object.
         """
+        prompt_template = prompt_template or self.default_prompt_template
+        if prompt_template is None:
+            return None
+
         if isinstance(prompt_template, PromptTemplate):
             return prompt_template
 
-        if not isinstance(prompt_template, str) or prompt_template not in self.prompt_templates:
-            raise ValueError(f"Prompt template {prompt_template} not supported")
+        if isinstance(prompt_template, str) and prompt_template in self.prompt_templates:
+            return self.prompt_templates[prompt_template]
 
-        return self.prompt_templates[prompt_template]
+        # if it's not a string or looks like a prompt template name
+        if not isinstance(prompt_template, str) or re.fullmatch(r"[-a-zA-Z0-9_]+", prompt_template):
+            raise ValueError(
+                f"{prompt_template} not supported, please select one of: {self.get_prompt_template_names()} or pass a PromptTemplate instance for prompting."
+            )
```

Comment:
Here we're sure that something is wrong:
Either it's an unexpected type or the string looks like a prompt template name but it isn't.